### PR TITLE
AP-809 Add loading state to button

### PR DESCRIPTION
**Pull request by justinanastos**
## Release Notes

- Add `loading` prop to `Button`
- Add `theme="grayscale"` option to `LoadingSpinner`
- Make `LoadingSpinner` motion linear
- Write preliminary docs page for `LoadingSpinner`

Contributes to issue https://apollographql.atlassian.net/browse/AP-809

### CODE REVIEW

File: src/Loaders/LoadingSpinner.tsx [hunk 1]
```diff
@@ -25,18 +25,9 @@ interface Props {
...
 }
 
-// The whole animation is exactly 5 seconds long.
-// Each rotation is 450 degrees, with a deceleration at
-// 420 degrees, and a reduced speed between 420-450.
-// 1st rotation: 60 deg and 90 deg
-// 2nd rotation: 150 deg and 180 deg
-// 3rd rotation: 240 deg and 270 deg
-// 4th rotation: 330 deg and 360 deg (restart loop)
 const SPIN = keyframes`
```
Comment (jgzuke) [inline src/Loaders/LoadingSpinner.tsx]: nbd: I didn't know we wanted to get rid of the fancy animation for all loaders, I thought it was only for loader icons. 👍assuming that switch did happen but wanted to confirm

Comment (justinanastos) [reply src/Loaders/LoadingSpinner.tsx]: Good question @jgzuke ; see this https://meteor.slack.com/archives/CFSCN3G7L/p1567213272006100

**Comment by justinanastos** [comment]
@jgzuke Good catch, we totally want a Props table here!

**Comment by apollo-bot2** [comment]
:rocket: PR was released in v2.4.0 :rocket:

**Pull request merged**